source/parser.hera

```diff
@@ -4914,6 +4922,16 @@ Init
       return [rest, after]
     }
 
+    // Ensure given Parameters structure isn't parsed as JSX by TypeScript,
+    // by replacing singleton TypeParameters <Foo> with <Foo,>.
+    // Modifies `parameters` in-place.
+    module.parametersNotJSX = function(parameters) {
+      const [tp] = parameters.children
+      if (tp && tp.type === "TypeParameters" && tp.parameters.length === 1) {
+        tp.parameters.push(",")
```

Comment:
Since this mutates in place it may cause a problem with some caching/backtracking situations. I'd recommend either moving the mutation to after the entire parse completes (no more backtracking so we're free to mutate) or returning a new `parameters` structure and re-assigning at the calling location.

Reply:
I'm new to the mutation after the entire parse completes. Could I just add some code to `processParams` say?

Reply:
Yes. Basically anything called inside `processProgram` (and so forth) is free to mutate the tree and nodes. Anything called in a handler shouldn't mutate anything that is not created from inside the handler itself since it may receive the same cached results multiple times.